econml/utilities.py

```diff
@@ -544,6 +546,33 @@ def check_input_arrays(*args, validate_len=True):
     return args
 
 
+def get_input_columns(X):
+    """Extracts column names from dataframe-like input object.
+
+    Currently supports column name extraction from pandas DataFrame and Series objects.
+
+    Parameters
+    ----------
+    X : array_like
+        Input array with column names to be extracted.
+
+    Returns
+    -------
+    cols: array-like or None
+        List of columns corresponding to the dataframe-like object.
+        None if the input array is not in the supported types.
+
+    """
+    # Type to column extraction function
+    type_to_func = {
+        pd.DataFrame: lambda x: x.columns.tolist(),
+        pd.Series: lambda x: [x.name]
+    }
+    if type(X) in type_to_func:
+        return type_to_func[type(X)](X)
+    return None
```

Comment:
Seems simpler without the dictionary:

```suggestion
    if type(X) == pd.DataFrame:
        return X.columns.tolist()
    elif type(X) == pd.Series:
        return [X.name]
    else:
        return None
```

Reply:
I left is a dictionary in case we will add other dataframes in the future. E.g. tensorflow, spark, lightgbm have their own dataframe-like structure that carry metadata. Hopefully, we will support all of them at some point.